util/headerreader/header_reader.go

```diff
@@ -88,13 +91,22 @@ var TestConfig = Config{
 }
 
 func New(client arbutil.L1Interface, config ConfigFetcher) *HeaderReader {
+	isParentChainArbitrum := false
+	codeAt, err := client.CodeAt(context.Background(), types.ArbSysAddress, nil)
+	if err != nil {
+		return nil
```

Comment:
Let's modify this function to take a context and return an error in this case. As-is it'll result in a nil pointer deref at the callsites if it fails. I checked all the callsites and they all have a `ctx` and an easy way to handle errors (the only non-standard one is `nitro.go` where you'll need to use `log.Crit` as in above `if err != nil` statements).

Reply:
done.